lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+
+В этом шаге Вы будете реализовывать `SentenceEncoder`.
+
+#### Шаг 4.1 Инициализатор класса
+
+Инициализатор класса `SentenceEncoder` не принимает аргументов.
+
+Внутри инициализатора необходимо вызвать инициализатор родительского класса.
+
+Таким образом, у объекта класса `SentenceEncoder` должны быть как минимум следующие поля: `_word2id`, `_id2word`.
+
+Интерфейс:   
+```py
+class SentenceEncoder(TextEncoder):
+    def __init__(self) -> None:
+        pass
+```  
+
+#### Шаг 4.2 Заполнение атрибутов класса. Метод `_learn_indices`
+
+Данный метод кодировщика принимает на вход кортеж из строк и соответствующим образом заполняет атрибуты `_word2id` и `_id2word`. Метод ничего не возвращает.
+Каждой строке должно быть присвоено целое число. Одной строке присваивается ровно одно уникальное число. **Минимальное число среди присвоенных чисел должно быть не меньше 1000.**
+
+Рассмотрим пример. Пусть дана следующая последовательность токенов: `('в', 'лесу', 'родилась', 'ёлочка', 'в', 'лесу', 'она', 'росла')`
+В этой последовательности всего 6 разных строк: `'в'`, `'лесу'`, `'она'`, `'родилась'`, `'росла'`, `'ёлочка'`. 
+Присвоим каждой из этих строк по одному уникальному числу больше 1000. Допустим, соответствие получается такое:
+
+* `'в'` - `1009`
+* `'лесу'` - `1001`
+* `'она'` - `1005`
+* `'родилась'` - `1101`
+* `'росла'` - `1061`
+* `'ёлочка'` - `1041`
+
+Тогда атрибут `_word2id` должен выглядеть следующим образом: `{'в': 1009, 'лесу': 1001, 'она': 1005, 'родилась': 1101, 'росла': 1061, 'ёлочка': 1041}`. 
+В `_id2word`, соответственно, ключи и значения меняются местами: `{1009: 'в', 1001: 'лесу', 1005: 'она', 1101: 'родилась', 1061: 'росла', 1041: 'ёлочка'}`.
+
+Если вы все сделали правильно, то длина данных атрибутов должна совпадать.
+
+**Важно**: в рамках данной лабораторной работы метод должен **дополнять** атрибуты `_word2id` и `_id2word`, а не переписывать их. 
+Модифицируйте поведение метода, чтобы этот момент учитывался.
+
+Интерфейс:   
+```py
+def _learn_indices(self, tokens: tuple[str, ...]) -> None:
+    pass
+```
+
+#### Шаг 4.3 Кодирование предложений. Метод `encode_sentences`
+
+Этот метод кодирует предложения, то есть заменяет каждый токен на число. 
+
+Метод принимает последовательность предложений. 
+
+Метод добавляет к каждому предложению его закодированную версию.
+
+**Важно**: метод ничего не возвращает.
+
+**Важно №2**: на входящих предложениях необходимо использовать метод `_learn_indicies`.
+
+**Важно №3**: необходимо использовать методы `set_attribute` и `get_attribute` для взаимодействия с атрибутами экземпляров `Sentence`.  
+
+Интерфейс:   
+```py
+def encode_sentences(self, tokens: tuple[Sentence, ...]) -> None:
+    pass
+```  
+
+### Шаг 5. Продемонстрировать выделение и кодирование предложений. Выполнение Шагов 1-5 соответствует 6 баллам
+
+Выделите из текста, хранящегося в переменной `text`, предложения и закодируйте их. Это необходимо сделать в файле `start.py`.
```

Comment:
Change the order: ... in file ... there is alreade pre-loaded text ... in variable ... Demonstrate encoding of sentences